Makefile

```diff
@@ -0,0 +1,53 @@
+####### Universal ######
+
+pip:
+	pip install -e .
+
+pip-test: pip
+	pip install -e.[test]
+
+
+unit-test:
+	pytest -sv tests/unit
+
+blackbox-smoke-prefix:
+	echo "hello blackbox!"
+	pwd
+	ls -l
+	ls -l sandbox
+	cd sandbox && docker-compose ps
+
+integration-test:
+	pytest -sv tests/integration
+
+
+###### Local Only ######
+
+# assumes installations of pipx, build and tox via:
+# `pip install pipx; pipx install build; pipx install tox`
+local-project-build:
+	pyproject-build
+
+# assumes a symbolic link: sandbox -> /cloned/repo/algorand/sandbox
+local-sandbox-test:
+	./sandbox/sandbox test
+
+local-blackbox-smoke: blackbox-smoke-prefix local-sandbox-test
+
+local-blackbox: local-blackbox-smoke integration-test
+
+# assumes act is installed, e.g. via `brew install act`:
+local-gh-simulate:
+	act
+
+###### Github Actions Only ######
+
+gh-sandbox-test:
+	# relax exit code condition because indexer returns 500 when last-round = 0
+	script -e -c "bash -x ./sandbox/sandbox test" || echo "finished ./sandbox test"
+
+gh-blackbox-smoke: blackbox-smoke-prefix gh-sandbox-test
+
+gh-blackbox: gh-blackbox-smoke integration-test
+
+.PHONY: pip-publish pip-test unit-test gh-blackbox
```

Comment:
@tzaffi `pip-publish` appears among `.PHONY` targets, but nowhere else.  I'm wondering if it's stale?

If not stale, can you help me understand the target's intent?

Reply:
Good catch. `.PHONY` ensures that make targets which don't produce a file always run, even in the case that the file with the same name as the target exists. At some point during bootstraping this repo, my make commands became impotent and introducing the `.PHONY`'s fixed it. I'm not sure which file was being shadowed. However, we should _never_ be having shadowing problems on github, because we should never check a shadowing file into git. So I'll remove for now. If the problem ever pops again, I'll investigate it a little more deeply.